File: abc081-b/js/main.js
function Main(input) {
    input = input.split("\n");
    const num_list = input[1].trim().split(' ').map(Number);

    num_list_or = num_list.reduce((a, b) => a | b);

    let count = 0;
    while (num_list_or % 2 === 0) {
        count++;
        num_list_or /= 2;
    }

    console.log(count);
}

Main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: abc081-b/cpp/main.cpp
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

int main()
{
    int length;
    cin >> length;
    vector<int> num_list(length);
    for (int &num : num_list)
    {
        cin >> num;
    }

    int num_list_or = accumulate(num_list.begin(), num_list.end(), 0, [](int a, int b)
                                 { return a | b; });

    int count = 0;
    while (num_list_or % 2 == 0)
    {
        num_list_or >>= 1;
        count++;
    }

    cout << count << endl;
}
